torchtitan/optimizer.py

```diff
@@ -26,6 +35,14 @@ def __init__(self, model_parts, optimizer_kwargs, name):
                 raise NotImplementedError(f"Optimizer {name} not added.")
             self.optimizers.append(optimizer)
 
+    def update_for_checkpoint(self, model):
```

Comment:
why do we need this method and `self.model = []`? can we just do `self.model = model_parts` in `__init__` and other things in save/load_state_dict?